src/dodal/devices/util/save_panda.py

```diff
@@ -0,0 +1,72 @@
+import os
+import sys
+from argparse import ArgumentParser
+from typing import cast
+
+from bluesky.run_engine import RunEngine
+from ophyd_async.core import Device, save_device
+from ophyd_async.panda import phase_sorter
+
+from dodal.beamlines import module_name_for_beamline
+from dodal.utils import make_all_devices
+
+
+def main():
+    """CLI Utility to save the panda configuration."""
+    parser = ArgumentParser(description="Save an ophyd_async panda to yaml")
+    parser.add_argument(
+        "--beamline", help="beamline to save from e.g. i03. Defaults to BEAMLINE"
+    )
+    parser.add_argument(
+        "device_name", help="name of the device e.g. panda", default="panda"
+    )
+    parser.add_argument("output_file", help="output filename")
+
+    # this exit()s with message/help unless args parsed successfully
+    args = parser.parse_args()
+
+    beamline = args.beamline
+    device_name = args.device_name
+    output_file = args.output_file
+
+    if beamline:
+        os.environ["BEAMLINE"] = beamline
+    else:
+        beamline = os.environ.get("BEAMLINE", None)
+
+    if not beamline:
+        sys.stderr.write("BEAMLINE not set and --beamline not specified\n")
+        return 1
+
+    _save_panda(beamline, device_name, output_file)
+
+    print("Done.")
+    return 0
+
+
+def _save_panda(beamline, device_name, output_file):
+    RE = RunEngine()
+    print("Creating devices...")
+    module_name = module_name_for_beamline(beamline)
+    devices, exceptions = make_all_devices(
```

Comment:
We only need to create the panda device here

Reply:
I was being lazy - we don't seem to have a nice function for creating only one named device, only one for creating everything. I will create one.